### PR TITLE
Check for a 'maxBlocks' config when prefetching a checkpoint

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

File: core/backend/src/CheckpointManager.ts [hunk 1]
```diff
@@ -208,19 +208,25 @@ export class V2CheckpointManager {
...
-        const logPrefetch = async (prefetch: CloudSqlite.CloudPrefetch) => {
-          const stopwatch = new StopWatch(`[${container.containerId}/${dbName}]`, true);
-          Logger.logInfo(loggerCategory, `Starting prefetch of ${stopwatch.description}`, { minRequests, maxRequests, timeout });
-          const done = await prefetch.promise;
-          Logger.logInfo(loggerCategory, `Prefetch of ${stopwatch.description} complete=${done} (${stopwatch.elapsedSeconds} seconds)`, { minRequests, maxRequests, timeout });
-        };
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-        logPrefetch(CloudSqlite.startCloudPrefetch(container, dbName, { minRequests, nRequests: maxRequests, timeout }));
+        const maxBlocks = getPrefetchConfig("maxBlocks", 500); // default size of 2GB. Assumes a checkpoint block size of 4MB.
+        if (dbStats?.totalBlocks !== undefined && dbStats?.totalBlocks <= maxBlocks) {
```
Comment (kabentley) [inline core/backend/src/CheckpointManager.ts:218]: 2nd "?" is not necessary, 2G seems pretty small. No?

Comment (nick4598) [reply core/backend/src/CheckpointManager.ts:218]: Oops, removed the 2nd "?". Yeah I would say 2G is probably too small. I wanted to leave a conservative number for the default, and maybe 2G would be the starting point. The number will go up through settings provided by backends (which is controlled by a feature flag).

**Review by kabentley — APPROVED** [review]

**Comment by nick4598** [comment]
@mergifyio backport release/3.x

**Comment by nick4598** [comment]
@mergifyio backport release/4.0.x

**Pull request merged**